app/app.go

```diff
@@ -383,19 +405,65 @@ func (a *App) Start() error {
 	return nil
 }
 
+func (a *App) startTrigger(id string, trg *triggerWrapper) (bool,error) {
+
+	statusInfo := trg.status
+	err := managed.Start(fmt.Sprintf("Trigger [ %s ]", id), trg.trg)
+	if err != nil {
+		if a.stopOnError {
+			return false, fmt.Errorf("error starting Trigger[%s] : %s", id, err)
+		}
+		log.RootLogger().Infof("Trigger [%s] failed to start due to error [%s]", id, err.Error())
+		statusInfo.Status = managed.StatusFailed
+		statusInfo.Error = err
+		log.RootLogger().Debugf("StackTrace: %s", debug.Stack())
+		trigger.PostTriggerEvent(trigger.FAILED, id)
+		return false, nil
+	} else {
+		statusInfo.Status = managed.StatusStarted
+		//logger.Infof("Trigger [ %s ]: Started", id)
+		version := ""
+		log.RootLogger().Debugf("Trigger [ %s ] has ref [ %s ] and version [ %s ]", id, trg.ref, version)
+		trigger.PostTriggerEvent(trigger.STARTED, id)
+	}
+
+	return true, nil
+}
+
+
 func (a *App) Stop() error {
 
 	logger := log.RootLogger()
 
 	logger.Info("Stopping Triggers...")
 
-	// Stop Triggers
-	for id, trg := range a.triggers {
+	lifecycleTriggers := make(map[string]*triggerWrapper)
+	normalTriggers := make(map[string]*triggerWrapper)
+
+	for id, trgW := range a.triggers {
+		if _, ok := trgW.trg.(LifecycleAware); ok {
+			lifecycleTriggers[id] = trgW
+		} else {
+			normalTriggers[id] = trgW
+		}
+	}
+
+	// Stop Normal Triggers
+	for id, trg := range normalTriggers {
 		_ = managed.Stop("Trigger [ "+id+" ]", trg.trg)
 		trg.status.Status = managed.StatusStopped
 		trigger.PostTriggerEvent(trigger.STOPPED, id)
 	}
 
+	// Stop Lifecycle Triggers
+	for id, trgW := range lifecycleTriggers {
+		lca, _ := trgW.trg.(LifecycleAware)
+		_ = lca.OnShutdown()
```

Comment:
We should throw onShutdown error back otherwise we don't know what happens for onShutDown flow

Reply:
I don't think we should throw it back, we should probably just log the errors and continue trying to shut everything else down

Reply:
Yes. Log error message also fine.